pkg/osquery/extension.go

```diff
@@ -589,6 +594,15 @@ func (e *Extension) nodeKey() string {
 // RequireReenroll clears the existing node key information, ensuring that the
 // next call to Enroll will cause the enrollment process to take place.
 func (e *Extension) RequireReenroll(ctx context.Context) {
+	// We cannot perform reenrollment via this extension for secretless installations,
+	// because reenrollment requires an enroll secret. Instead, launcher will have to reenroll
+	// via localserver.
+	if e.isSecretless {
+		e.slogger.Log(ctx, slog.LevelWarn,
+			"RequireReenroll called for secretless installation, will not reenroll",
+		)
+		return
```

Comment:
I think what we do here is an _interesting_ question. Maybe ignoring is good enough? If we knew enough, we might want to treat it as an unenroll event?

Reply:
I support kicking the unenroll/reenroll can forward, yes. 

I wonder if we want the extension to _not_ send things if there's no node key. I suspect there are artfacts here of when we were mostly just osquery, and the usual enroll path was to have osquery do something, and get the re-enroll event.